src/lib/timing/index.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+/**
+ * Runtime profiling
+ */
+
+/**
+ * Levels of reporting
+ * @type {Object}
+ */
+const levels = {
+  warn: 1,
+  info: 2,
+  debug: 3
+};
+
+/**
+ * Runtime settings
+ * @type {Object}
+ */
+const settings = {
+  reportHandler: null,
+  currentLevel: levels.warn
+};
+
+/**
+ * Handles the reporting when a check 'fails'
+ * @param {Function} callback            Callback function to handle the report
+ * @param {Integer}  [level=levels.warn] Minimum level to report on
+ */
+function report(callback, level = levels.warn) {
+  if (typeof callback !== 'function') {
+    throw new TypeError('First argument must be a function');
+  }
+
+  settings.reportHandler = callback;
+  settings.currentLevel = level;
+}
+
+/**
+ * Converts result of process.hrtime() in milliseconds
+ * @param {Array}  hrResult The result to convert
+ * @return {Float}          Result as milliseconds
+ */
+function hrtimeToMs(hrResult) {
+  return (hrResult[0] * 1000) + (hrResult[1] / 1e6);
+}
+
+/**
+ * Creates a new check
+ * @param {String}  action         Name of the action to measure
+ * @param {Integer} infoThreshold  Millisecond threshold to emit info
+ * @param {Integer} warnThreshold  Millisecond threshold to emit warning
+ */
+function Check(action, infoThreshold, warnThreshold) {
```

Comment:
Yes. It will work, since i use the `new` keyword which will create an object and return it.